ftw/testbrowser/tests/test_forms.py

```diff
@@ -386,3 +387,43 @@ def test_simple_textarea_does_not_break(self, browser):
             '</form>')
         form = browser.fill({'text': 'some text'})
         self.assertEqual({'text': 'some text'}, dict(form.values))
+
+    @browsing
+    def test_find_submit_button_tag(self, browser):
+        browser.open_html(
+            '<form>'
+            '<button type="submit">blubb</button>'
+            '</form>')
+        form = browser.forms['form-0']
+        self.assertEquals(1, len(form.find_submit_buttons()))
+
+    @browsing
+    def test_find_submit_button_tag_by_label(self, browser):
+        browser.open_html(
+            '<form>'
+            '<button type="submit">blubb</button>'
+            '</form>')
+        form = browser.forms['form-0']
+        button = form.find_button_by_label('blubb')
+        self.assertTrue(button)
+        self.assertEquals('submit', button.type)
+
+    @browsing
+    def test_find_submit_button_tag_click(self, browser):
+        browser.open_html(
+            '<form action="http://nohost/plone">'
+            '<button type="submit">blubb</button>'
+            '</form>')
+        form = browser.forms['form-0']
+        button = form.find_button_by_label('blubb')
+        button.click()
+
+    @browsing
+    def test_find_submit_button_tag_in_request(self, browser):
+        browser.open_html(
+            '<form action="http://nohost/plone" method="POST">'
+            '<button name="gugus" type="submit">blubb</button>'
+            '</form>')
+        form = browser.forms['form-0']
+        form.submit()
+        self.assertIn('name="gugus"', browser.previous_request[1]['url'].get_data())
```

Comment:
I didn't realize at first that you are asserting the multipart POST data here.
I'm also missing the value (I guess it is an empty string?).

I would prefer to see more of the sent values. This can be easily achieved by actually posting to the view `test-form-result` [1], which simply returns the `request.form` [2].
See other tests above [3] as an example.

[1] https://github.com/4teamwork/ftw.testbrowser/blob/504bd158a74ecbbbfba50a600ffdb42ad8229e58/ftw/testbrowser/tests/views/test-form.pt#L10
[2] https://github.com/4teamwork/ftw.testbrowser/blob/504bd158a74ecbbbfba50a600ffdb42ad8229e58/ftw/testbrowser/tests/views/views.py#L13
[3] https://github.com/4teamwork/ftw.testbrowser/blob/504bd158a74ecbbbfba50a600ffdb42ad8229e58/ftw/testbrowser/tests/test_forms.py#L229-L233